src/Microsoft.ML.OnnxTransform/DnnImageFeaturizerTransform.cs

```diff
@@ -0,0 +1,105 @@
+        /// <param name="modelFactory">An extension method on the <see cref="DnnImageModelSelector"/> that creates a chain of two
+        /// <see cref="OnnxScoringEstimator"/>s with specific models included in a package together with that extension method.
+        /// For an example, see Microsoft.ML.DnnImageFeaturizer.ResNet18 </param>
+        /// <param name="input">Input column name.</param>
+        /// <param name="output">Output column name.</param>
+        public DnnImageFeaturizerEstimator(IHostEnvironment env, Func<DnnImageModelSelector, IHostEnvironment, string, string, EstimatorChain<OnnxTransform>> modelFactory, string input, string output)
+        {
+            _modelChain = modelFactory(new DnnImageModelSelector(), env, input, output);
+        }
+
+        /// <summary>
+        /// Note that OnnxEstimator which this is based on is a trivial estimator, so this does not do any actual training,
+        /// just verifies the schema.
+        /// </summary>
+        public TransformerChain<OnnxTransform> Fit(IDataView input)
+        {
+            return _modelChain.Fit(input);
+        }
+
+        public SchemaShape GetOutputSchema(SchemaShape inputSchema)
+        {
+            return _modelChain.GetOutputSchema(inputSchema);
+        }
+    }
+
+    public static class DnnImageFeaturizerStaticExtensions
+    {
+        private sealed class OutColumn : Vector<float>
+        {
+            public PipelineColumn Input { get; }
+
+            public OutColumn(Vector<float> input, Func<DnnImageModelSelector, IHostEnvironment, string, string, EstimatorChain<OnnxTransform>> model)
+                : base(new Reconciler(model), input)
+            {
+                Input = input;
+            }
+        }
+
+        private sealed class Reconciler : EstimatorReconciler
+        {
+            private readonly Func<DnnImageModelSelector, IHostEnvironment, string, string, EstimatorChain<OnnxTransform>> _model;
+
+            public Reconciler(Func<DnnImageModelSelector, IHostEnvironment, string, string, EstimatorChain<OnnxTransform>> model)
+            {
+                _model = model;
+            }
+
+            public override IEstimator<ITransformer> Reconcile(IHostEnvironment env,
+                PipelineColumn[] toOutput,
+                IReadOnlyDictionary<PipelineColumn, string> inputNames,
+                IReadOnlyDictionary<PipelineColumn, string> outputNames,
+                IReadOnlyCollection<string> usedNames)
+            {
+                Contracts.Assert(toOutput.Length == 1);
+
+                var outCol = (OutColumn)toOutput[0];
+                return new DnnImageFeaturizerEstimator(env, _model, inputNames[outCol.Input], outputNames[outCol]);
+            }
+        }
+
+        public static Vector<float> DnnImageFeaturizer(this Vector<float> input, Func<DnnImageModelSelector, IHostEnvironment, string, string, EstimatorChain<OnnxTransform>> model)
```

Comment:
>DnnImageFeaturizer [](start = 36, length = 18)

Public function needs very detailed doc string.

Reply:
Thanks, added!

---
In reply to: [232436619](https://github.com/dotnet/machinelearning/pull/1447#discussion_r232436619) [](ancestors = 232436619)